src/Uno.Analyzers/UnoDoNotDisposeNativeViews.cs

```diff
@@ -0,0 +1,148 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Uno.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class UnoDoNotDisposeNativeViews : DiagnosticAnalyzer
+	{
+		internal const string Title = "Do not dispose native views";
+		internal const string MessageFormat = "Do not call Dispose() on {0}";
+		internal const string Description = "Calling Dispose on native controls can cause undefined behaviors and runtime crashes.";
+		internal const string Category = "Reliability";
+
+		internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+#pragma warning disable RS2008 // Enable analyzer release tracking
+			"Uno0002",
+#pragma warning restore RS2008 // Enable analyzer release tracking
+			Title,
+			MessageFormat,
+			Category,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true,
+			description: Description,
+			helpLinkUri: "https://aka.platform.uno/UNO0002"
+		);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
```

Comment:
Do we really want to report the diagnostic in generated code?

Reply:
I'm not sure. There may be generators that create that pattern...